src/Package/Impl/DataInspect/VariableView.xaml.cs

```diff
@@ -57,16 +71,76 @@ public partial class VariableView : UserControl, IDisposable {
             e.Handled = true;
         }
 
-        private void OnGlobalEnvironmentEvaluation(DebugEvaluationResult result) {
-            var wrapper = new EvaluationWrapper(result);
+        private void EnvironmentProvider_EnvironmentChanged(object sender, REnvironmentChangedEventArgs e) {
+            int selectedIndex = 0;
 
-            var rootNodeModel = new VariableNode(_settings, wrapper);
+            var currentItem = EnvironmentComboBox.SelectedItem as REnvironment;
+            if (currentItem != null && !e.Environments[0].FrameIndex.HasValue) {
+                int index = e.Environments.FindIndex((env) => env.Name == currentItem.Name);
+                selectedIndex = index == -1 ? selectedIndex : index;
+            }
+
+            EnvironmentComboBox.ItemsSource = e.Environments;
+            EnvironmentComboBox.SelectedIndex = selectedIndex;
+        }
 
-            VsAppShell.Current.DispatchOnUIThread(
-                () => {
-                    EnvironmentName.Text = GlobalEnvironmentName;
-                    _rootNode.Model = rootNodeModel;
-                });
+        private void EnvironmentComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e) {
+            if (!object.Equals(EnvironmentComboBox.ItemsSource, _defaultEnvironments) && e.AddedItems.Count > 0) {
+                var env = e.AddedItems[0] as REnvironment;
+                if (env != null) {
+                    SetRootModelAsync(env).SilenceException<Exception>().DoNotWait();
```

Comment:
I don't like this pattern of silencing any exception during method execution. What exceptions do you expect?